Improve how to deploy a collection (#172)
OpenTermsArchive/docs: content/collections/how-to/manage-custom-terms-type.md

---
title: Manage a custom terms type
weight: 6
---

# How to manage a custom terms type

When tracking terms and conditions across different services, you might encounter situations where a service uses terms types that aren't yet supported by Open Terms Archive. 
This guide will help you handle these custom terms types effectively, whether you're working on a new collection or expanding an existing one.

## Understanding the validation

Open Terms Archive validates terms types to ensure consistency and quality across collections. When you encounter a validation error like:

```shell
   1) Service declarations validation
       <service_name>
         valid declaration schema:
      Error:

 data/terms must be equal to one of the allowed values (in entire file)
```

> This means the terms type you're trying to use isn't in the official list of supported types.

Reviewer: Should this section really be in a “how-to” guide? 🤔

> This means the terms type you're trying to use isn't in the official list of supported types.

Reviewer: Is there no way to improve the error message rather than explaining it in documentation?

> ## Before going further

Reviewer: ```suggestion
## Double-check potential synonyms
```

Before proceeding with using a custom terms type, please double-check that the type you're considering doesn't already exist in the supported terms types list. Different services often use different names for the same type of terms. For instance, "Terms and Conditions" might be called "Terms of Use" or "Terms of Service" by different services.

Review the [supported types list](https://github.com/OpenTermsArchive/terms-types/blob/main/termsTypes.json) carefully to ensure the terms type you want to track is not already supported under a different name. If it does, use the associated type in your declaration. If you are sure it's not supported, you can proceed with the following steps. If you're unsure, you can ask for help in the [Open Terms Archive community]({{< relref "community/how-to/join" >}}). 

## Long-term solution

> The recommended approach is to contribute your new terms type to the official list. This helps maintain consistency across collections and enables better comparison of terms across different services.

Reviewer: ```suggestion
The recommended approach is to contribute your new terms type to the official list. This enables collections interoperability and comparison of terms across different services.
```

> You can propose your new terms type for inclusion in the official list of supported types via the [dedicated repository](https://github.com/OpenTermsArchive/terms-types), following the [contribution process](https://github.com/OpenTermsArchive/terms-types/blob/main/CONTRIBUTING.md#add-new-terms-types).

Reviewer: ```suggestion
You can suggest new terms type for the official list via the dedicated [contribution process](https://github.com/OpenTermsArchive/terms-types/blob/main/CONTRIBUTING.md#add-new-terms-types).
```

This process has the following benefits:

- Your terms type becomes available to all collections
- You don't need to maintain a custom fork
- It helps build a more comprehensive and useful shared list

> - Enables better comparison of terms across different services

Reviewer: Not sure this should be part of a how-to.

> This solution can take some time to have your terms type to be approved, so if you need a faster solution, you can proceed with the following temporary solutions.

Reviewer: ```suggestion
Inclusion in the official terms types list has a delay for consensus building. In the meantime, you can proceed with the following temporary solutions.
```
